nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -95,12 +95,19 @@ void LatticeMotionTable::initMotionModel(
 
     if (prev_start_angle != new_primitive.start_angle) {
       motion_primitives.push_back(primitives);
+
+      // adding direction change index, used only if reversing is allowed
+      direction_change_indices.push_back(static_cast<unsigned int>(primitives.size()));
```

Comment:
I don't believe this works. The reversing primitives are not added [until here](https://github.com/ros-planning/navigation2/blob/7bb4954cdb3c8b486d10ef1e8318dd1861993465/nav2_smac_planner/src/node_lattice.cpp#L121-L147), they're not stored in the file themselves. It seems better that `getMotionPrimitives` returns the `idx` of the reversing where it is populated instead. 

How did you test this & working?

Reply:
the current implementation definitely works: 
- the method used to test was to compare the direction change index from the old implementation to the current one (the parameter file was updated to use the smac lattice planner & reversing was set to true). The gazebo simulation was used with random start and end points (manually done) to check if the direction change index is correct.
- The core logic of the implementation is that  in the function getMotionPrimitives a list is first created with **all forward primitives** for the current heading , then the reverse primitives are appended. This implies the `direction_change index` is nothing but the number of primitives in the forward direction.

The reason I added a section as future work is that in the issue  #3966 it was suggested the calculation to be done at initialization (this is how its implemented now) but I felt we could also "save memory " by not storing the values but just returning the index in getMotionPrimitives function.